schema.ts

```diff
@@ -1321,9 +1348,62 @@ function checkSchemaPropertyIsRequired<K extends string, L extends string, T ext
   );
 }
 
+/**
+ * Normalizes a schema property key into PascalCase. This interprets "."s as accessing object properties
+ * and "[]" as accessing array items.
+ */
 export function normalizeSchemaKey(key: string): string {
-  // Colons cause problems in our formula handling.
-  return pascalcase(key).replace(/:/g, '_');
```

Comment:
What happens if you were cross-docing a column called, say, "Foo [bar]", will that break as soon as this gets checked in? I think before it would get normalized to "FooBar", but now it will remain "Foo [bar]" and so I think both it will get a new column name when resyncing, and it may also just cause the model to barf in some way. Seems like we should test this first.

Assuming it breaks something, we may want to make the detection of path-like strings much more conservative here, and use a regex to only consider strings that are syntactically valid paths, without whitespace and whose brackets only contain numbers. Even still we may break people who use periods in cross-doc column names.

Reply:
yeah let me add a bunch more tests for these edge cases. I think in the logic on the coda side it should always look for an exact object key match before it even tries to do any pathing